deltacat/compute/compactor/compaction_session.py

```diff
@@ -285,33 +261,20 @@ def _execute_compaction_round(
     assert hash_bucket_count is not None and hash_bucket_count > 0, \
         f"Expected hash bucket count to be a positive integer, but found " \
         f"`{hash_bucket_count}`"
-
-    # rehash the primary key index if necessary
-    if round_completion_info:
-        logger.info(f"Round completion file contents: {round_completion_info}")
-        # the previous primary key index is compatible with the current, but
-        # will need to be rehashed if the hash bucket count has changed
-        if hash_bucket_count != old_hash_bucket_count:
-            # TODO(draghave): manually test the path after prior primary key 
-            #   index was already built
-            round_completion_info = pki.rehash(
-                round_robin_opt_provider,
-                compaction_artifact_s3_bucket,
-                source_partition_locator,
-                round_completion_info,
-                hash_bucket_count,
-                max_parallelism,
-                records_per_primary_key_index_file,
-                delete_prev_primary_key_index,
-            )
-
+    hb_pre_end = time.time()
+    hb_start = time.time()
     # parallel step 1:
     # group like primary keys together by hashing them into buckets
+    if not round_completion_info and rebase_source_partition_locator:
+        # generate a rc_info for hb and materialize to tell whether a delta is src or destination in case of rebase
+        dest_delta_locator = DeltaLocator.of(partition_locator=rebase_source_partition_locator, stream_position=None)
+        round_completion_info = RoundCompletionInfo.of(None, dest_delta_locator, None, 0, None)
```

Comment:
Please check comment on `hash_bucket`, we could avoid this empty object creation just with rebase source delta locator.

Reply:
will leave this as enhancement